Assets/_BForBoss/_Leaderboard/Scripts/Username/InputUsernameViewBehaviour.cs

```diff
@@ -0,0 +1,52 @@
+using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+
+namespace BForBoss
+{
+    /// <summary>
+    /// Placeholder view just to input the name
+    /// </summary>
+    public class InputUsernameViewBehaviour : MonoBehaviour
+    {
+        [SerializeField] private TMP_InputField _usernameField = null;
+        [SerializeField] protected TMP_Text _infoSettingsLabel = null;
+        [SerializeField] private Button _saveUsernameButton = null;
+        protected InputUsernameViewModel _viewModel = null;
+        
+        public virtual void Initialize(InputUsernameViewModel viewModel = null)
+        {
+            _viewModel = viewModel ?? new InputUsernameViewModel();
+            BindViewModel();
+            _saveUsernameButton.onClick.AddListener(() =>
+            {
+                _viewModel.SetUserName(_usernameField.text);
+            });
+        }
+
+        protected virtual void BindViewModel()
+        {
+            _viewModel.OnFailure += ShowFailedText;
+        }
+        
+        private void ShowFailedText()
+        {
+            _infoSettingsLabel.text =
+                "Something is wrong with your username, try another one. (No Blank Text or names over 20 characters";
```

Comment:
[nitpick[ Maybe we should have a public int getter in `InputUsername`  to get the characterlimit amount. cause if we change that number in that class then this hardcoded string would be wrong!

```suggestion
                _infoSettingsLabel.text = $"Something is wrong with your username, try another one. (No Blank Text or names over {InputUserName.CharacterLimit} characters";
```